subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/screenshot/ScreenshotCapturerImpl.kt

```diff
@@ -44,7 +44,11 @@ class ScreenshotCapturerImpl(private val reportFileProvider: ReportFileProvider)
         compressFormat: Bitmap.CompressFormat,
         quality: Int
     ): Result<File?> {
-        return reportFileProvider.provideReportDir().flatMap { dir ->
+        return Result.tryCatch {
+            val reportDir = reportFileProvider.provideReportDir()
+            reportDir.mkdirs()
```

Comment:
We duplicate this knowledge in provider's clients.
It's error prone.

Can we hide it inside the provider?

Reply:
fixed